File: src/main/java/org/example/expert/domain/comment/CommentQueryDslRepository.java
package org.example.expert.domain.comment;

import org.example.expert.controller.comment.dto.response.CommentResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface CommentQueryDslRepository {
    Page<CommentResponse> getCommentsWithUserByTodoId(Long todoId, Pageable pageable);
}

File: src/main/java/org/example/expert/domain/comment/CommentQueryDslRepository.kt
package org.example.expert.domain.comment

import org.example.expert.controller.comment.dto.response.CommentInfoRespDto
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface CommentQueryDslRepository {

    fun getCommentsWithUserByTodoId(todoId: Long, pageable: Pageable): Page<CommentInfoRespDto>
}